docs/plugins/creating-plugin.rst

```diff
@@ -64,11 +64,14 @@ to ensure your plugin's extraordinary quality.
 
 Besides the way you are creating plugins (based on our skeleton or on your own), there are a few naming conventions that should be followed:
 
- * Bundle class must have a ``Plugin`` suffix instead of ``Bundle`` in its name (e.g. InvoicePlugin).
+ * Bundle class must have a ``Sylius`` prefix, and a ``Plugin`` suffix (instead of ``Bundle``) in its name (e.g. ``SyliusInvoicePlugin``).
```

Comment:
I'd make it *vendor name* + `Sylius` + *actual name* + `Plugin`. The reasoning behind it is to distinct official Sylius plugins from the community made ones. `SyliusInvoicePlugin` uses `sylius_invoice` as config root name, whereas `AcmeSyliusInvoicePlugin` uses `acme_sylius_invoice.`

Reply:
True.. Downside is that e.g. in my case using my own name we get to something like this: `StefanDoornSyliusInvoicePlugin`.. sounds silly :-) The vendor is already in the namespace and the repository is in another vendor, isn't that enough to make people (developers only) understand what the source of it is?

Reply:
These are only two files, bundle one and extension one, it's most likely to be an one time thing. The issue here is to have distinct config keys.

For Composer package name and Github repos it's already solved by prefixing with `vendor-name/`.

Reply:
Ah that way. Ok, I'll check it. Alternative could be to adjust the trait to adjust the config key I think based on PSR4 namespace.

Reply:
Is there really a need to distinct official Sylius plugins from the community made plugins? There should be one rule for them all IMO. 
And `Sylius` + `name` + `Plugin` seems simple and enough.

Reply:
Using vendor names in package names / configuration keys is a common practice, I can't think of Symfony bundle which does not follow this convention.

Let's just take our most common plugin, CMS one, there's one from Lakion and one from BitBag. If we stay with `SyliusCmsPlugin`, the configuration key will always be `sylius_cms`, which might cause confusion. And then FriendsOfSylius or Sylius itself wants to unify them... creating another one with the same configuration key.

Reply:
I see the point. Is it possible to do this by adjusting the trait? What I'm trying is to get the name of the repo and the class name aligned, but having diversified configs keys :)

Reply:
As for the currently existing trait, changing its behaviour will be a BC break and having two of them might be misleading.

Reply:
Makes sense, not an option. Could you (@pamil, @pjedrzejewski, @CoderMaggie) let me know what the naming convention regarding this will be? Then I will implement accordingly.

Reply:
I guess we can end up with:

Composer package name: `vendor-name/sylius-something-plugin`
Bundle / extension classname: `VendorNameSyliusSomethingPlugin` / `VendorNameSyliusSomethingExtension` - namespace does not matter, probably in `VendorName\SyliusSomethingPlugin`